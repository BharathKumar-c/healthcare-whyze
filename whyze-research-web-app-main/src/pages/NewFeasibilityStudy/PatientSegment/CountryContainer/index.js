import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'antd';
import { Star, StarFilled, Tick } from '../../../../assets';
import { Text } from '../../../../components/basic';
import { totalPatientFormatter } from '../../../../utils/calculations';
import { PlusOutlined } from '@ant-design/icons';

export default function CountryContainer({
  isPinned,
  countryObj,
  onUnPinCountry,
  onPinCountry,
}) {
  const { country, count } = countryObj;
  return (
    <div className="country-container">
      <Row>
        <Col span={18}>
          <Row className="">
            <Col
              className="country-container-column"
              span={4}
              style={{ cursor: 'pointer' }}
            >
              {isPinned ? (
                <StarFilled
                  onClick={() => onUnPinCountry(country)}
                  className="favorite-isFilled"
                />
              ) : (
                <Star
                  onClick={() => onPinCountry(country)}
                  className="favorite-notFilled"
                />
              )}
            </Col>
            <Col
              span={20}
              style={{
                whiteSpace: 'nowrap',
                overflow: 'hidden',
                textOverflow: 'ellipsis',
              }}
            >
              <Text
                className={
                  isPinned
                    ? 'patient-segment-pinned-title'
                    : 'patient-segment-title'
                }
                text={country ? country : 'Country not found'}
              />
            </Col>
          </Row>
        </Col>
        <Col span={6}>
          <Row style={{ alignItems: 'flex-end' }}>
            <Col
              span={14}
              className={
                isPinned ? 'no-of-patients' : 'country-container-no-of-patients'
              }
            >
              {totalPatientFormatter(count)}
            </Col>
            <Col span={10} style={{ cursor: 'pointer' }}>
              {isPinned ? (
                <Tick
                  onClick={() => console.log('tick', country)}
                  className="Tick-icon"
                />
              ) : (
                <PlusOutlined
                  onClick={() => console.log('add', country)}
                  className="Pluse-Icon"
                />
              )}
            </Col>
          </Row>
        </Col>
      </Row>
    </div>
  );
}

CountryContainer.propTypes = {
  isPinned: PropTypes.bool,
  countryObj: PropTypes.objectOf(
    PropTypes.shape({
      country: PropTypes.object,
      count: PropTypes.number,
    }),
  ),
  onUnPinCountry: PropTypes.func,
  onPinCountry: PropTypes.func,
};

CountryContainer.defaultProps = {
  isPinned: false,
  countryObj: {
    country: '',
    count: 0,
  },
  onUnPinCountry: () => {},
  onPinCountry: () => {},
};
