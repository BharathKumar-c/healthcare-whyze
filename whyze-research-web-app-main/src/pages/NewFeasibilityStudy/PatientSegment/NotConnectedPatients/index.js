import React from 'react';
import PropTypes from 'prop-types';
import { UserDelete } from '../../../../assets';
import { Popover } from '../../../../components/hoc';
import { InfoCircleFilled } from '@ant-design/icons';

export function NotConnectedPatients(props) {
  const { noOfNotConncetedPatients } = props;

  const content = (
    <div className="content">
      Additional Whyze patients in site catchment areas, but not yet connected
      to these sites.
    </div>
  );
  return (
    <div className="not-connected-patients-container">
      <div className="alignment">
        <UserDelete />
        <div className="no-of-patients-text">{noOfNotConncetedPatients}</div>
      </div>
      <div className="alignment">
        <div className="sites-text">Not connected to sites</div>
        <Popover
          id="not-connected-patients"
          trigger="hover"
          content={content}
          placement="bottomRight"
        >
          <InfoCircleFilled />
        </Popover>
      </div>
    </div>
  );
}

NotConnectedPatients.propTypes = {
  noOfNotConncetedPatients: PropTypes.number,
};

NotConnectedPatients.defaultProps = {
  noOfNotConncetedPatients: 0,
};
