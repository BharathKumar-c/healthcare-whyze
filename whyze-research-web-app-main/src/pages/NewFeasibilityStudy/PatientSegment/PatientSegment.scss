@import '../../../common_styles/common_variable.scss';

.patient-segment {
    padding: 25px;
    background: linear-gradient(180deg, rgba(179, 211, 49, 0.1) 0%, rgba(179, 211, 49, 0.02) 100%);
    border-radius: 12px;
    width: 100%;
    height: 100%;

    &-title {
        @include font(400, 16px, 140%, $secondary-text-color !important);
        margin-bottom: 5px;

        &-small {
            @include font(400, 15px, 16px, $secondary-text-color !important);
            margin-bottom: 5px;
        }
    }

    .site-country-container {
        margin-bottom: 16px;
    }

    .pinned-country-container {
        height: 14%;
        margin-bottom: 20px;
    }

    .country-title {
        margin-top: 3%;
    }

    .search-field {
        margin-left: 4%;
        cursor: pointer;
    }

    .error-message {
        width: 88%;
        text-align: center;
        background-image: url('../../../assets/EmptyContentBackground.png');
        background-repeat: no-repeat;
        background-position: center;
        background-size: cover;
        margin-top: 5px;
        padding: 10px;
    }

    &-pinned-title {
        @include font(400, 16px, 140%, $primary-text-color !important);
    }

    .sub-title {
        @include font(400, 18px, 140%, $primary-text-color);
    }

    .site-container {
        background: $white-color;
        border-radius: 8px;
        min-width: 132px;
        margin-right: 15px;
        padding: 10px;

        &-icon {
            width: 40px;
            height: 40px;
            background: $tertiary-background-color;
            border-radius: 24px;
            display: flex;
            align-items: center;
            justify-content: center;

            &-grey-bg {
                width: 40px;
                height: 40px;
                border-radius: 24px;
                display: flex;
                align-items: center;
                justify-content: center;
                background: #F6F7F9;
            }
        }

    }

    .no-of-patients {
        @include font(700, 18px, 140%, $primary-text-color);
    }

    .countries-list {
        overflow-y: scroll;
        max-height: 150px;

        &::-webkit-scrollbar {
            display: none;
        }
    }

    .country-container {
        padding: 7px 0;
        border-bottom: 1px solid rgba(45, 51, 57, 0.08);

        &-column{
            display: flex;
            align-items: center;
        }
        &-no-of-patients {
            @include font(700, 18px, 140%, $secondary-text-color);
        }
    }

    .not-connected-patients-container {
        background: $white-color;
        border-radius: 8px;

        .alignment {
            display: flex;
            align-items: center;
            padding: 10px;
            flex-direction: row;
        }

        .no-of-patients-text {
            @include font(700, 18px, 140%, $secondary-text-color);
            margin-left: 10px;
        }

        .sites-text {
            @include font(400, 12px, 16px, $secondary-text-color);
            flex-grow: 1;
        }

    }

    .selected-countries {

        &-title {
            @include font(400, 18px, 140%, $primary-text-color !important);
        }

        &-component {
            align-items: center;
            padding: 6px 12px 3px;
            background: #F6F7F9;
            border-radius: 16px;
            margin-top: 10px;

            &-border {
                align-items: center;
                padding: 6px 12px 3px;
                margin-top: 10px;
                border-bottom: 1px solid rgba(45, 51, 57, 0.08);
            }
        }

        &-location-icon {
            path {
                fill: rgba(45, 51, 57, 0.8);
            }
        }

        &-text {
            overflow: hidden;
            white-space: nowrap;
            text-overflow: ellipsis;
            @include font(400, 16px, 140%, $secondary-text-color);
        }

        &-no-of-patients {
            display: flex !important;
            justify-content: center;
            @include font(700, 16px, 140%, $primary-text-color)
        }
    }

    .site-overview {
        margin-bottom: 15px;

        &-title {
            @include font(400, 16px, 140%, $secondary-text-color);
        }

        &-container {
            display: flex;
            justify-content: space-evenly;
            align-items: center;
            padding: 8px;
            background: $white-color;
            border-radius: 8px;
            margin-top: 10px;
        }

        &-alignment {
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
        }

        &-vertical-alignment {
            display: flex;
            justify-content: space-between;
            align-items: center;
        }

        &-icon {
            margin-right: 10px;

            path {
                fill: #2D3339;
            }
        }

        &-value {
            @include font(400, 18px, 140%, $tertiary-text-color);
        }

        &-text {
            @include font(400, 12px, 16px, $secondary-text-color);
        }

        &-divider {
            border-left: 1px solid rgba(45, 51, 57, 0.08);
            margin-left: 5px;
        }
    }
}

#not-connected-patients,
.ant-popover-inner {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
    padding: 4px 8px;
    width: 220px;
    background: #138686;
    border-radius: 4px;

    .content {
        @include font(400, 12px, 16px, $white-color);
    }

}