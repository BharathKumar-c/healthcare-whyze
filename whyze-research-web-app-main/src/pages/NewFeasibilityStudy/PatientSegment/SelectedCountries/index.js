import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row, Switch } from 'antd';
import { Location } from '../../../../assets';
import { Text } from '../../../../components/basic';

export default function SelectedCountries({ countriesList, isShowSwitch }) {
  return (
    <div className="selected-countries-container">
      <Text className="selected-countries-title" text="Selected countries" />
      {countriesList.map((country, index) => (
        <Row
          key={index}
          className={
            isShowSwitch
              ? 'selected-countries-component-border'
              : 'selected-countries-component'
          }
        >
          <Col span={isShowSwitch ? 24 : 15}>
            <Row>
              <Col span={4}>
                <Location className="selected-countries-location-icon" />
              </Col>
              <Col className="selected-countries-text" span={20}>
                {country?.country}
              </Col>
            </Row>
          </Col>
          <Col span={9}>
            {!isShowSwitch && (
              <Row>
                <Col className="selected-countries-no-of-patients" span={15}>
                  {country?.count}
                </Col>
                <Col span={9}>
                  <Switch size="small" />
                </Col>
              </Row>
            )}
          </Col>
        </Row>
      ))}
    </div>
  );
}

SelectedCountries.propTypes = {
  countriesList: PropTypes.array,
  isShowSwitch: PropTypes.bool,
};

SelectedCountries.defaultProps = {
  countriesList: [],
  isShowSwitch: false,
};
