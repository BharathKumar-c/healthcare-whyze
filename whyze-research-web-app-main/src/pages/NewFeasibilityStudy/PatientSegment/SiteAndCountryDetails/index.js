import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'antd';
import { Text } from '../../../../components/basic';
import { InfoCircleFilled } from '@ant-design/icons';
import { Popover } from '../../../../components/hoc';

export default function SiteAndCountryDetails({
  icon,
  value,
  title,
  isInfo,
  tab,
}) {
  const content = (
    <div className="content">
      Additional Whyze patients in site catchment areas, but not yet connected
      to these sites.
    </div>
  );
  return (
    <div className="site-container">
      <Row align="middle" wrap style={{ marginBottom: '10px' }}>
        <Col span={12}>
          <div
            className={
              isInfo ? 'site-container-icon-grey-bg' : 'site-container-icon'
            }
          >
            {icon}
          </div>
        </Col>
        <Col span={12}>
          <Text className="no-of-patients" text={value} />
        </Col>
      </Row>
      <Row align="middle">
        <Col span={isInfo ? 22 : 24}>
          <Text
            className={
              tab === 'feasibility' || tab === 'location'
                ? 'patient-segment-title'
                : 'patient-segment-title-small'
            }
            text={title}
          />
        </Col>
        {isInfo && (
          <Col span={2}>
            <Popover
              id="not-connected-patients"
              trigger="hover"
              content={content}
              placement="bottomRight"
            >
              <InfoCircleFilled />
            </Popover>
          </Col>
        )}
      </Row>
    </div>
  );
}

SiteAndCountryDetails.propTypes = {
  value: PropTypes.number,
  title: PropTypes.string,
  icon: PropTypes.node,
  isInfo: PropTypes.bool,
  tab: PropTypes.string,
};

SiteAndCountryDetails.defaultProps = {
  value: '',
  title: '',
  icon: null,
  isInfo: false,
  tab: '',
};
