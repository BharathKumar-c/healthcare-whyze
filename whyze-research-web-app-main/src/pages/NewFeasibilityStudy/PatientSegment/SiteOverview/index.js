import { Divider } from 'antd';
import React from 'react';
import { Contact, Initiated, Sites } from '../../../../assets';
import { Text } from '../../../../components/basic';
import PropTypes from 'prop-types';
export default function SiteOverview({
  preferredSitesCount,
  sitesInitiatedCount,
  sitesContactedCount,
}) {
  return (
    <div className="site-overview">
      <Text className="site-overview-title" text="Sites overview" />
      <div className="site-overview-container">
        <div className="site-overview-alignment">
          <div className="site-overview-vertical-alignment">
            <div className="site-overview-icon">
              <Sites />
            </div>
            <Text className="site-overview-value" text={preferredSitesCount} />
          </div>
          <Text className="site-overview-text" text="Sites" />
        </div>
        <Divider className="site-overview-divider" type="vertical" />
        <div className="site-overview-alignment">
          <div className="site-overview-vertical-alignment">
            <div className="site-overview-icon">
              <Contact />
            </div>
            <Text className="site-overview-value" text={sitesContactedCount} />
          </div>
          <Text className="site-overview-text" text="Contacted" />
        </div>
        <Divider className="site-overview-divider" type="vertical" />
        <div className="site-overview-alignment">
          <div className="site-overview-vertical-alignment">
            <div className="site-overview-icon">
              <Initiated />
            </div>
            <Text className="site-overview-value" text={sitesInitiatedCount} />
          </div>
          <Text className="site-overview-text" text="Initiated" />
        </div>
      </div>
    </div>
  );
}

SiteOverview.propTypes = {
  preferredSitesCount: PropTypes.number,
  sitesInitiatedCount: PropTypes.number,
  sitesContactedCount: PropTypes.number,
};
SiteOverview.defaultProps = {
  preferredSitesCount: 0,
  sitesInitiatedCount: 0,
  sitesContactedCount: 0,
};
