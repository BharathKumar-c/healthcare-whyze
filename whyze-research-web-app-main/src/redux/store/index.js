import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
