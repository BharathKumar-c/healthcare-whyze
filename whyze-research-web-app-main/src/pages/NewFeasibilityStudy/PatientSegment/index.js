import { SearchOutlined } from '@ant-design/icons';
import { Col, Row } from 'antd';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import {
  Location,
  Sites,
  Search,
  OutlinedClose,
  UserDelete,
  UserAdd,
} from '../../../assets';
import { Input, Text } from '../../../components/basic';
import CountryContainer from './CountryContainer';
import { NotConnectedPatients } from './NotConnectedPatients';
import SelectedCountries from './SelectedCountries';
import SiteAndCountryDetails from './SiteAndCountryDetails';
import './PatientSegment.scss';
import SiteOverview from './SiteOverview';
import { errorConstants } from '../../../constant';
import { useSelector } from 'react-redux';
import store from '../../../redux/store';

export default function PatientSegment({
  totalNoOfPatients,
  countriesList,
  tab,
  totalCountry,
  siteCount,
  noOfNotConncetedPatients,
}) {
  const [isSearch, setIsSearch] = useState(false);
  const [countries, setCountries] = useState(countriesList || []);
  const [isPinnedCountries, setIsPinnedCountries] = useState(false);
  const [filteredCountriesList, setFilteredCountriesList] = useState([]);
  const { activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const { segments } = store.getState().feasibilityStudyReducer;

  const countriesLength = segments[activeSegment.index];
  const {
    preferredSites,
    sitesContacted,
    sitesInitiated,
    selectedCountries,
    sites,
    totalConnectedNotConnected,
  } = segments[activeSegment.index];
  const selectedCountriesList = selectedCountries?.map((ele) => ({
    country: ele.name,
    count: ele.count,
  }));

  let connectedSites = preferredSites?.reduce((a, b) => {
    return a + b.connected || 0;
  }, 0);

  let notConnectedSites = preferredSites?.reduce((a, b) => {
    return a + b.not_connected_sites || 0;
  }, 0);
  let totalConnectedAndNotConnected = connectedSites + notConnectedSites;

  useEffect(() => {
    setCountries(countriesList);
  }, [countriesList]);

  const topCountriesList =
    countries &&
    countries
      .sort((a, b) => Number(b.count) - Number(a.count))
      .filter((ele, idx) => !ele.isFavourite && idx <= 3);

  const handleChange = (event) => {
    if (event.target.value.length > 0) {
      setFilteredCountriesList(
        countries.filter(
          (ele) =>
            ele.country
              ?.toLowerCase()
              .includes(event.target.value.toLowerCase()) && !ele.isFavourite,
        ),
      );
    } else {
      setFilteredCountriesList([]);
    }
  };

  const handlePinCountry = (country) => {
    if (isSearch) {
      // To add a favorite country when opening the search filter
      const newFilteredCountries = [...filteredCountriesList];
      const deleteCountryIndex = newFilteredCountries.findIndex(
        (ele) => ele.country === country,
      );
      newFilteredCountries.splice(deleteCountryIndex, 1);

      setFilteredCountriesList(newFilteredCountries);
      const newCountries = [...countries];
      newCountries.forEach((ele) => {
        if (ele.country === country) {
          ele.isFavourite = true;
        }
      });
      setIsPinnedCountries(true);
      setCountries(newCountries);
    } else {
      const newCountries = [...countries];
      newCountries.forEach((ele) => {
        if (ele.country === country) {
          ele.isFavourite = true;
        }
      });
      setIsPinnedCountries(true);
      setCountries(newCountries);
    }
  };

  const handleUnPinCountry = (country) => {
    const newCountries = [...countries];
    newCountries.forEach((ele) => {
      if (ele.country === country) {
        ele.isFavourite = false;
      }
    });
    setCountries(newCountries);
    setIsPinnedCountries(countries.findIndex((ele) => ele.isFavourite) !== -1);
  };

  const handleCloseSearchField = () => {
    setFilteredCountriesList([]);
    setIsSearch(false);
  };

  useEffect(() => {
    // Finding if favorite countries are there on the list
    setIsPinnedCountries(countries.findIndex((ele) => ele.isFavourite) !== -1);
  }, []);

  return (
    <div className="patient-segment">
      <Row>
        <Text className="patient-segment-title" text="Patient segment" />
      </Row>
      <Row>
        <Text
          headerType={2}
          text={
            tab == 'feasibility'
              ? totalNoOfPatients?.feasibility || 0
              : tab === 'sites' ||
                (tab === 'location' && totalNoOfPatients?.location)
              ? totalNoOfPatients?.location
              : tab === 'preferredList'
              ? totalConnectedAndNotConnected || 0
              : !totalNoOfPatients?.feasibility
              ? 0
              : totalNoOfPatients?.feasibility
          }
        />
      </Row>
      {!(tab === 'feasibility' || tab === 'location') && (
        <Row>
          <Text className="patient-segment-title" text="Breakdown" />
        </Row>
      )}
      <Row className="site-country-container">
        <Col span={12}>
          <SiteAndCountryDetails
            icon={
              tab === 'feasibility' || tab === 'location' ? (
                <Sites />
              ) : (
                <UserAdd />
              )
            }
            value={
              tab === 'feasibility'
                ? siteCount?.feasibility
                : tab === 'location' &&
                  sites &&
                  Object.keys(sites)?.length !== 0
                ? siteCount?.location
                : tab === 'location' &&
                  (!sites || Object.keys(sites)?.length === 0) &&
                  selectedCountries?.length === 0
                ? siteCount?.feasibility
                : tab === 'location' &&
                  (!sites || Object.keys(sites)?.length === 0)
                ? 0
                : tab === 'sites'
                ? totalConnectedNotConnected?.connectedPatientCount || 0
                : tab === 'preferredList'
                ? connectedSites || 0
                : !siteCount?.feasibility
                ? 0
                : siteCount?.feasibility
            }
            title={
              tab === 'feasibility' || tab === 'location'
                ? 'Sites'
                : 'Connected to sites'
            }
            tab={tab}
          />
        </Col>
        <Col span={12}>
          <SiteAndCountryDetails
            icon={
              tab === 'feasibility' || tab === 'location' ? (
                <Location />
              ) : (
                <UserDelete />
              )
            }
            value={
              tab === 'feasibility'
                ? totalCountry || 0
                : tab === 'location' &&
                  countriesLength?.selectedCountries?.length > 0
                ? countriesLength?.selectedCountries?.length
                : tab === 'sites'
                ? totalConnectedNotConnected?.notConnectedPatientCount || 0
                : tab === 'preferredList'
                ? notConnectedSites || 0
                : !totalCountry
                ? 0
                : totalCountry
            }
            title={
              tab === 'feasibility' || tab === 'location'
                ? 'Countries'
                : 'Not connected to sites'
            }
            isInfo={!(tab === 'feasibility' || tab === 'location')}
            tab={tab}
          />
        </Col>
      </Row>
      {tab === 'feasibility' && (
        <>
          {isPinnedCountries && (
            <div className="pinned-country-container">
              <Text className="sub-title" text="Your pinned countries" />
              {countries.map((country, i) => (
                <>
                  {country.isFavourite && (
                    <CountryContainer
                      key={i}
                      isPinned
                      countryObj={country}
                      onUnPinCountry={handleUnPinCountry}
                    />
                  )}
                </>
              ))}
            </div>
          )}
          <Row className="country-title">
            <Col span={20}>
              {isSearch ? (
                <Input
                  name={'SearchCountries'}
                  placeholder="Search countries ..."
                  prefix={<SearchOutlined />}
                  handleInputChange={handleChange}
                />
              ) : (
                <Text
                  className="sub-title"
                  text={
                    countriesLength?.country?.length > 1
                      ? `Top ${countriesLength?.country?.length} countries`
                      : countriesLength?.country?.length === 1
                      ? `Top ${countriesLength?.country?.length} country`
                      : ''
                  }
                />
              )}
            </Col>
            <Col span={3} className="search-field">
              {isSearch ? (
                <OutlinedClose
                  onClick={handleCloseSearchField}
                  className="close-search-icon"
                />
              ) : (
                <Search onClick={() => setIsSearch(true)} />
              )}
            </Col>
          </Row>
          <div className="countries-list">
            {!isSearch &&
              topCountriesList.map((country, i) => (
                <>
                  {!country.isFavourite && (
                    <CountryContainer
                      key={i}
                      countryObj={country}
                      onPinCountry={handlePinCountry}
                    />
                  )}
                </>
              ))}
            {isSearch &&
              filteredCountriesList.length > 0 &&
              filteredCountriesList.map((country, i) => (
                <CountryContainer
                  key={i}
                  countryObj={country}
                  onPinCountry={handlePinCountry}
                />
              ))}
            {isSearch && filteredCountriesList.length === 0 && (
              <div className="error-message">
                <Text
                  className="patient-segment-pinned-title"
                  text={errorConstants.patientNullError}
                />
              </div>
            )}
          </div>
        </>
      )}
      {tab === 'location' && noOfNotConncetedPatients ? (
        <NotConnectedPatients />
      ) : null}
      {tab === 'preferredList' && (
        <SiteOverview
          preferredSitesCount={preferredSites?.length}
          sitesContactedCount={sitesContacted?.length}
          sitesInitiatedCount={sitesInitiated?.length}
        />
      )}
      {(tab === 'sites' || tab === 'preferredList') && (
        <SelectedCountries
          countriesList={selectedCountriesList}
          isShowSwitch={tab === 'preferredList'}
        />
      )}
    </div>
  );
}

PatientSegment.propTypes = {
  totalNoOfPatients: PropTypes.object,
  countriesList: PropTypes.array,
  tab: PropTypes.string,
  totalCountry: PropTypes.number,
  connetedSites: PropTypes.number,
  siteCount: PropTypes.object,
  noOfNotConncetedPatients: PropTypes.number,
};

PatientSegment.defaultProps = {
  totalNoOfPatients: {},
  countriesList: [],
  tab: '',
  totalCountry: 0,
  connetedSites: 0,
  siteCount: {},
  noOfNotConncetedPatients: 0,
};
