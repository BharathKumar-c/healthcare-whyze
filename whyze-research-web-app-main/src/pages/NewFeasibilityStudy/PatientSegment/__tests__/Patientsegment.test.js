import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../redux/store';
import React from 'react';
import PatientSegment from '..';
import userEvent from '@testing-library/user-event';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
const democountryList = [
  { Country: 'United Kingdom' },
  { Country: 'Albania', count: '1' },
  { Country: 'Brazil', count: '4' },
  { Country: 'Ireland', count: '4' },
];
describe('test cases for PatientSegment ', () => {
  test('should create snapshots', () => {
    const { container } = renderWithContext(<PatientSegment />);
    expect(container).toMatchSnapshot();
  });
  test('should test title', () => {
    const { container } = renderWithContext(<PatientSegment />);
    expect(
      container.getElementsByClassName(
        'ant-typography patient-segment-title',
      )[0].firstChild.textContent,
    ).toBe('Patient segment');
  });
  test('should Corrct icon render', () => {
    const { container } = renderWithContext(<PatientSegment />);
    const PatientLogo = container.getElementsByClassName('site-container-icon');
    expect(PatientLogo[0].firstChild.textContent).toBe('UserAdd.svg');
  });
  test('should user delete icon render', () => {
    const { container } = renderWithContext(<PatientSegment />);
    const PatientLogo = container.getElementsByClassName(
      'site-container-icon-grey-bg',
    );

    expect(PatientLogo[0].firstChild.textContent).toBe('UserDelete.svg');
  });
  test('sould isSearch icon is click and search country using input tag', () => {
    const { container, getByText } = renderWithContext(
      <PatientSegment tab={'feasibility'} />,
    );
    const SearchIcon = container.getElementsByClassName('search-field');
    userEvent.click(SearchIcon[0].firstChild);
    expect(getByText(/Oops, we cant find that country/i)).toBeInTheDocument();
    const countryInput = container.querySelector(
      'input[name="SearchCountries"]',
    );
    expect(countryInput.value).toMatch('');
    fireEvent.change(countryInput, { target: { value: 'ireland' } });
    expect(countryInput.value).toMatch('ireland');
  });
  test('should isSearch icon not  click ', () => {
    const { container } = renderWithContext(
      <PatientSegment tab={'feasibility'} countriesList={democountryList} />,
    );
    const testTest = container.getElementsByClassName('sub-title');

    expect(testTest[0].firstChild.textContent).toBe('Top 4 countries');
    const pinCountry = container.getElementsByClassName('country-container');
    fireEvent.click(pinCountry[0].firstChild);
    expect(testTest.length).toBe(1);
  });
  test('should pin and unpin the favorite country to the table', () => {
    const { container, getByText } = renderWithContext(
      <PatientSegment tab={'feasibility'} countriesList={democountryList} />,
    );
    const SearchIcon = container.getElementsByClassName('search-field');
    userEvent.click(SearchIcon[0].firstChild);
    // get countries-list
    const countryInput = container.querySelector(
      'input[name="SearchCountries"]',
    );

    expect(countryInput.value).toMatch('');
    fireEvent.change(countryInput, { target: { value: 'ireland' } });
    expect(countryInput.value).toMatch('ireland');
    const favoriteNotFilled =
      container.getElementsByClassName('favorite-notFilled');

    expect(getByText(/ireland/i)).toBeInTheDocument();

    //pin selected country
    fireEvent.click(favoriteNotFilled[0].firstChild);
    expect(container.getElementsByClassName('country-container').length).toBe(
      1,
    );
    const Pinnedelements = container.getElementsByClassName(
      'pinned-country-container',
    );

    expect(Pinnedelements[0].firstChild.textContent).toBe(
      'Your pinned countries',
    );
    expect(Pinnedelements.length).toBe(1);
    //unpin selected country
    const favoriteFilled =
      container.getElementsByClassName('favorite-isFilled');
    fireEvent.click(favoriteFilled[0].firstChild);
    expect(container.getElementsByClassName('country-container').length).toBe(
      0,
    );
  });
  test('should test handlechnage in search country', () => {
    const { container } = renderWithContext(
      <PatientSegment tab={'feasibility'} countriesList={democountryList} />,
    );
    const SearchIcon = container.getElementsByClassName('search-field');
    userEvent.click(SearchIcon[0].firstChild);
    // countries-list
    const countryInput = container.querySelector(
      'input[name="SearchCountries"]',
    );
    fireEvent.change(countryInput, { target: { value: 'ireland' } });
    expect(countryInput.value).toBe('ireland');
  });
  test('should input notBe filled', () => {
    const { container } = renderWithContext(
      <PatientSegment tab={'feasibility'} countriesList={democountryList} />,
    );
    const SearchIcon = container.getElementsByClassName('search-field');
    userEvent.click(SearchIcon[0].firstChild);
    // countries-list
    const countryInput = container.querySelector(
      'input[name="SearchCountries"]',
    );
    fireEvent.change(countryInput, '');
    expect(countryInput.textContent).toBe('');
  });
  test('should onclick search input close', () => {
    const { container } = renderWithContext(
      <PatientSegment tab={'feasibility'} countriesList={democountryList} />,
    );
    const SearchIcon = container.getElementsByClassName('search-field');
    userEvent.click(SearchIcon[0].firstChild);
    const CloseIcon = container.getElementsByClassName('close-search-icon');
    fireEvent.click(CloseIcon[0].firstChild);
  });
});
