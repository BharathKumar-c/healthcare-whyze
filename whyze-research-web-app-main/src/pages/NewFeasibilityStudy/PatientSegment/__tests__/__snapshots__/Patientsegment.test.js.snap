// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for PatientSegment  should create snapshots 1`] = `
<div>
  <div
    class="patient-segment"
  >
    <div
      class="ant-row"
    >
      <span
        class="ant-typography patient-segment-title"
      >
        Patient segment
      </span>
    </div>
    <div
      class="ant-row"
    >
      <h2
        class="ant-typography"
      >
        0
      </h2>
    </div>
    <div
      class="ant-row"
    >
      <span
        class="ant-typography patient-segment-title"
      >
        Breakdown
      </span>
    </div>
    <div
      class="ant-row site-country-container"
    >
      <div
        class="ant-col ant-col-12"
      >
        <div
          class="site-container"
        >
          <div
            class="ant-row ant-row-middle"
            style="margin-bottom: 10px;"
          >
            <div
              class="ant-col ant-col-12"
            >
              <div
                class="site-container-icon"
              >
                <svg>
                  UserAdd.svg
                </svg>
              </div>
            </div>
            <div
              class="ant-col ant-col-12"
            >
              <span
                class="ant-typography no-of-patients"
              >
                0
              </span>
            </div>
          </div>
          <div
            class="ant-row ant-row-middle"
          >
            <div
              class="ant-col ant-col-24"
            >
              <span
                class="ant-typography patient-segment-title-small"
              >
                Connected to sites
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-12"
      >
        <div
          class="site-container"
        >
          <div
            class="ant-row ant-row-middle"
            style="margin-bottom: 10px;"
          >
            <div
              class="ant-col ant-col-12"
            >
              <div
                class="site-container-icon-grey-bg"
              >
                <svg>
                  UserDelete.svg
                </svg>
              </div>
            </div>
            <div
              class="ant-col ant-col-12"
            >
              <span
                class="ant-typography no-of-patients"
              >
                0
              </span>
            </div>
          </div>
          <div
            class="ant-row ant-row-middle"
          >
            <div
              class="ant-col ant-col-22"
            >
              <span
                class="ant-typography patient-segment-title-small"
              >
                Not connected to sites
              </span>
            </div>
            <div
              class="ant-col ant-col-2"
            >
              <span
                aria-label="info-circle"
                class="anticon anticon-info-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="info-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
